spark2/src/main/java/org/apache/iceberg/spark/SparkFilters.java

```diff
@@ -135,6 +136,13 @@ public static Expression convert(Filter filter) {
 
         case NOT:
           Not notFilter = (Not) filter;
+          Filter sparkChild = notFilter.child();
+          // Special case to use the Iceberg NOT_STARTS_WITH operator for
+          // Not(StringStartsWith) to allow filter push down.
+          if (sparkChild instanceof StringStartsWith) {
+            StringStartsWith castedChild = (StringStartsWith) sparkChild;
+            return notStartsWith(castedChild.attribute(), castedChild.value());
+          }
```

Comment:
I don't think this is necessary. It's fine to convert `not(startsWith(...))` to the same structure. The important thing is that when we rewrite to remove `not`, we can produce a `notStartsWith` instead of failing.

We probably don't need to touch Spark at all.

Reply:
Yeah I removed this and tested everything and it seems fine to go.